android-core/src/main/java/com/mparticle/modernization/identity/MParticleIdentity.kt

```diff
@@ -0,0 +1,94 @@
+package com.mparticle.modernization.identity
+
+import com.mparticle.identity.IdentityApiRequest
+import com.mparticle.identity.IdentityApiResult
+import com.mparticle.identity.MParticleUser
+import com.mparticle.modernization.MParticleCallback
+import com.mparticle.modernization.core.MParticleComponent
+import org.jetbrains.annotations.NotNull
+import org.jetbrains.annotations.Nullable
+
+interface MParticleIdentity : MParticleComponent {
+
+    /**
+     * Login
+     * @param callback
+     */
+    fun login(@NotNull callback: IdentityCallback)
+
+    /**
+     * Logout
+     * @param callback
+     */
+    fun logout(@NotNull callback: IdentityCallback)
+
+    /**
+     * Returns the user by id or the current user if [mpId] is null
+     * @param mpId if null returns the current user, if not the user by id
+     * @param callback
+     */
+    fun getUser(@Nullable mpId: Long?, @NotNull callback: IdentityCallback)
+
+    /**
+     * Return all users
+     * @param callback
+     */
+    fun getUsers(@NotNull callback: MParticleCallback<List<MParticleUser>, Unit>)
+
+    /**
+     * Returns the user by id or the current user if [mpId] is null
+     * @param mpId if null returns the current user, if not the user by id
+     * @return identityApiResult
+     */
+    suspend fun getUser(@Nullable mpId: Long?): IdentityApiResult
+
+    /**
+     * Return all users
+     * @return mParicleUsers in a list
+     */
+    suspend fun getUsers(): List<MParticleUser>
+
+    /**
+     * Login
+     * @return identityApiResult
+     */
+    suspend fun login(): IdentityApiResult
+
+    /**
+     * Logout
+     * @return identityApiResult
+     */
+    suspend fun logout(): IdentityApiResult
+}
+
+internal interface InternalIdentity : MParticleIdentity {
```

Comment:
I'm a bit confused as to what `InternalIdentity` is.  Is there a reason the `MParticleIdentity` interface above does not include modify and identify, but the `InternalIdentity` interface does?

Reply:
So in general, the concept of having internal extensions to public interfaces is to limit what we make public while still having the functionality for internal use. In this particular case, I'm not sure why these are internal though...

Reply:
Mostrly read/write access to stuff.. there are operations that should be internal only.